### PR TITLE
refactor: remove simple_load_balancer

### PR DESCRIPTION
**Pull request by levy5307**
remove simple_load_balancer, and make the interface server_load_balancer simple.

### Manual Test
1. start onebox with 5 replica servers and 3 meta server2
2. create a table name with "temp", we can see create table process is ok.
3. kill 3 replica servers
4. use ddd_diagnose to see the ddd partitions
```>>> ddd_diagnose
Total 1 ddd partitions:

(1) 2.3
    config: ballot(6), last_committed(8)
    ----
    dropped[0]: node(10.231.57.100:34801), drop_time(2021-09-01 18:56:19.670), alive(false), collected(false), ballot(-1), last_committed(-1), last_prepared(-1)
    dropped[1]: node(10.231.57.100:34802), drop_time(2021-09-01 18:56:29.673), alive(false), collected(false), ballot(-1), last_committed(-1), last_prepared(-1)  <== the secondary latest
    dropped[2]: node(10.231.57.100:34803), drop_time(2021-09-01 18:56:43.560), alive(false), collected(false), ballot(-1), last_committed(-1), last_prepared(-1)  <== the latest
    ----
    last_drops[0]: node(10.231.57.100:34801)
    last_drops[1]: node(10.231.57.100:34802)  <== the secondary latest
    last_drops[2]: node(10.231.57.100:34803)  <== the latest
    ----
    ddd_reason: the last dropped node(10.231.57.100:34803) haven't come back yet

Proposed count: 0/1

Diagnose ddd done.
```

### CODE REVIEW

**Review by Smityz — APPROVED** [review]

File: src/meta/meta_options.cpp [hunk 1]
```diff
@@ -161,6 +161,11 @@ void meta_options::initialize()
     _lb_opts.only_move_primary = dsn_config_get_value_bool(
         "meta_server", "only_move_primary", false, "only try to make the primary balanced by move");
 
+    partition_guardian_type = dsn_config_get_value_string("meta_server",
+                                                          "partition_guardian_type",
+                                                          "partition_guardian",
+                                                          "partition guardian provider");
+
```
Comment (hycdong) [inline src/meta/meta_options.cpp:168]: One of the `partition_guardian_type` is "partition_guardian", does it have another type?

Comment (levy5307) [reply src/meta/meta_options.cpp:168]: There is another type name dummy_partition_guardian in unit test.

File: src/meta/test/balancer_validator.cpp [hunk 2]
```diff
@@ -35,17 +35,16 @@ namespace replication {
...
 {
+    meta_service svc;
+    partition_guardian guardian(&svc);
     pc_status ps = pc_status::invalid;
     node_state *ns;
 
     configuration_proposal_action act;
     while (ps != pc_status::healthy) {
-        ps = lb->cure({&apps, &nodes}, pc.pid, act);
+        ps = guardian.cure({&apps, &nodes}, pc.pid, act);
```
Comment (Shuo-Jia) [inline src/meta/test/balancer_validator.cpp:47]: The unit test is `guardian test`, but not the file name `balancer_validator` mean, you can also separate it from this file

Comment (levy5307) [reply src/meta/test/balancer_validator.cpp:47]: I don't think so. It`s a function used in balancer_validator, which used to test greedy load balancer

**Review by Shuo-Jia — APPROVED** [review]

**Review by hycdong — APPROVED** [review]

**Pull request merged**